internal/kafka/internal/services/kafka_tls_certificate_management/kafka_tls_certificate_management_config.go

```diff
@@ -0,0 +1,131 @@
+package kafka_tls_certificate_management
+
+import (
+	"fmt"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/environments"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+	"github.com/caddyserver/certmagic"
+	"github.com/go-playground/validator/v10"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+)
+
+const (
+	inMemoryStorageType            = "in-memory"
+	secureStorageType              = "secure-storage"
+	fileStorageType                = "file"
+	manualCertificateManagement    = "manual"
+	automaticCertificateManagement = "automatic"
+)
+
+var validStorageTypes = []string{inMemoryStorageType, fileStorageType, secureStorageType}
+var validCertificateManagementStrategies = []string{manualCertificateManagement, automaticCertificateManagement}
+
+type KafkaTLSCertificateManagementConfig struct {
+	CertificateAuthorityEndpoint         string
+	CertificateManagementStrategy        string
+	StorageType                          string
+	EnableKafkaExternalCertificate       bool
+	ManualCertificateManagementConfig    ManualCertificateManagementConfig
+	AutomaticCertificateManagementConfig AutomaticCertificateManagementConfig
+}
+
+type ManualCertificateManagementConfig struct {
+	KafkaTLSCert     string
+	KafkaTLSCertFile string `validate:"required"`
+	KafkaTLSKey      string
+	KafkaTLSKeyFile  string `validate:"required"`
+}
+
+type AutomaticCertificateManagementConfig struct {
+	RenewalWindowRatio        float64 `validate:"gte=0,lte=1"`
+	EmailToSendNotificationTo string  `validate:"required,email"`
+	AcmeIssuerAccountKeyFile  string  `validate:"required"`
+	AcmeIssuerAccountKey      string
+	MustStaple                bool
+}
+
+func NewCertificateManagementConfig() *KafkaTLSCertificateManagementConfig {
```

Comment:
Same comment as https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1531/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#r1102826227

Reply:
Sorry, the link doesn't get me to the previous comment. Do you mind giving me a gist of the previous comment again? Thanks

Reply:
Ah right. Thanks. No, I've not considered it. It's possible to do it but I just ("this being an enhancement of an old PR") took inspiration on the already existing PRs. 
I am open to have this changed / revisited in follow up enhancements PRs.

Reply:
Changing this in future PR will prove difficult as once introduced in stage/prod it won't be trivial to migrate